File: apps/ios/Barns/BarnsApp/Domain/UseCases/Patterns/GetPatternDetailUseCase.swift
/// Returns a single wall-greenery pattern by id, if present.
struct GetPatternDetailUseCase {
    private let repository: PatternRepository

    init(repository: PatternRepository) {
        self.repository = repository
    }

    func execute(id: WallGreenPattern.ID) async throws -> WallGreenPattern? {
        try await repository.wallGreenPattern(id: id)
    }
}

File: apps/android/app/src/main/java/com/barns/app/domain/usecase/patterns/GetPatternDetailUseCase.kt
package com.barns.app.domain.usecase.patterns

import com.barns.app.domain.model.WallGreenPattern
import com.barns.app.domain.repository.PatternRepository

/**
 * Returns a single wall-greenery pattern by id, if present.
 */
class GetPatternDetailUseCase(
    private val repository: PatternRepository,
) {
    suspend fun execute(id: String): WallGreenPattern? = repository.wallGreenPattern(id)
}
